### ISSUE TITLE
cannot reduce width and maintain height with dsize, and causes crash

### ISSUE THREAD

**Issue by pennyscan** [mpv-player/mpv#4606]
mpv-x86_64-20170708-git-7c1db05 Windows 10 pro 1607 latest

cannot reduce video width only with dsize[=w:h:aspect-method:r:aspect]

can make video wider, but cannot make thinner, height changes instead if try
I have tried adjusting all available options in dsize

see video of behaviour  - https://youtu.be/7B-C7vt_Raw
see video for desired behaviour  - https://youtu.be/YnobYs_dVR8

also, MPV crashes about once every ten times I change dsize - see video

I can provide any more data or testing if help
(this is clean portable install straight from original download folder, just added 2 line input.conf)

crash report

Version=1
EventType=APPCRASH
EventTime=131444102054274334
ReportType=2
Consent=1
UploadTime=131444102054963231
ReportIdentifier=20607b35-67aa-11e7-9c19-60f81dc92483
IntegratorReportIdentifier=5a633f8d-2d26-4dad-bb0d-15a6962f75c2
NsAppName=mpv.exe
AppSessionGuid=00000bf8-0001-0059-bd1f-a3d9b6fbd201
TargetAppId=W:000615c23613ffbfca2caca2d6e18aeb4aac00000000!0000361784d0a9206d4044d3fabf1b3970f70863d3fb!mpv.exe
TargetAppVer=1970//01//01:00:00:00!25266f9!mpv.exe
BootId=4294967295
Response.BucketId=9dc6d82fa6e8b4da4cb40aca31af5ba8
Response.BucketTable=4
Response.LegacyBucketId=120775567644
Response.type=4
Sig[0].Name=Application Name
Sig[0].Value=mpv.exe
Sig[1].Name=Application Version
Sig[1].Value=2.0.0.0
Sig[2].Name=Application Timestamp
Sig[2].Value=00000000
Sig[3].Name=Fault Module Name
Sig[3].Value=nvwgf2umx.dll
Sig[4].Name=Fault Module Version
Sig[4].Value=21.21.13.7654
Sig[5].Name=Fault Module Timestamp
Sig[5].Value=5864ff96
Sig[6].Name=Exception Code
Sig[6].Value=c0000005
Sig[7].Name=Exception Offset
Sig[7].Value=0000000000025e4f
DynamicSig[1].Name=OS Version
DynamicSig[1].Value=10.0.14393.2.0.0.256.48
DynamicSig[2].Name=Locale ID
DynamicSig[2].Value=2057
DynamicSig[22].Name=Additional Information 1
DynamicSig[22].Value=a701
DynamicSig[23].Name=Additional Information 2
DynamicSig[23].Value=a701895748692fefe88264cf060f4b4e
DynamicSig[24].Name=Additional Information 3
DynamicSig[24].Value=d911
DynamicSig[25].Name=Additional Information 4
DynamicSig[25].Value=d911b239ed0042dc7cc1a26fbcebdc7f
UI[2]=C:\Users\-\Documents\mpv.exe
UI[3]=mpv has stopped working
UI[4]=Windows can check online for a solution to the problem.
UI[5]=Check online for a solution and close the program
UI[6]=Check online for a solution later and close the program
UI[7]=Close the program
LoadedModule[0]=C:\Users\-\Documents\mpv.exe
LoadedModule[1]=C:\WINDOWS\SYSTEM32\ntdll.dll
LoadedModule[2]=C:\WINDOWS\System32\KERNEL32.DLL
LoadedModule[3]=C:\WINDOWS\System32\KERNELBASE.dll
LoadedModule[4]=C:\WINDOWS\System32\ADVAPI32.dll
LoadedModule[5]=C:\WINDOWS\System32\msvcrt.dll
LoadedModule[6]=C:\WINDOWS\System32\sechost.dll
LoadedModule[7]=C:\WINDOWS\System32\RPCRT4.dll
LoadedModule[8]=C:\WINDOWS\System32\GDI32.dll
LoadedModule[9]=C:\WINDOWS\System32\gdi32full.dll
LoadedModule[10]=C:\WINDOWS\SYSTEM32\d3d9.dll
LoadedModule[11]=C:\WINDOWS\System32\USER32.dll
LoadedModule[12]=C:\WINDOWS\SYSTEM32\AVRT.dll
LoadedModule[13]=C:\WINDOWS\SYSTEM32\dwmapi.dll
LoadedModule[14]=C:\WINDOWS\System32\win32u.dll
LoadedModule[15]=C:\WINDOWS\System32\combase.dll
LoadedModule[16]=C:\WINDOWS\System32\ole32.dll
LoadedModule[17]=C:\WINDOWS\System32\ucrtbase.dll
LoadedModule[18]=C:\WINDOWS\System32\OLEAUT32.dll
LoadedModule[19]=C:\WINDOWS\System32\msvcp_win.dll
LoadedModule[20]=C:\WINDOWS\System32\bcryptPrimitives.dll
LoadedModule[21]=C:\WINDOWS\System32\SHELL32.dll
LoadedModule[22]=C:\WINDOWS\System32\cfgmgr32.dll
LoadedModule[23]=C:\WINDOWS\SYSTEM32\OPENGL32.dll
LoadedModule[24]=C:\WINDOWS\SYSTEM32\Secur32.dll
LoadedModule[25]=C:\WINDOWS\System32\windows.storage.dll
LoadedModule[26]=C:\WINDOWS\System32\powrprof.dll
LoadedModule[27]=C:\WINDOWS\SYSTEM32\GLU32.dll
LoadedModule[28]=C:\WINDOWS\SYSTEM32\DDRAW.dll
LoadedModule[29]=C:\WINDOWS\System32\shlwapi.dll
LoadedModule[30]=C:\WINDOWS\System32\kernel.appcore.dll
LoadedModule[31]=C:\WINDOWS\SYSTEM32\DCIMAN32.dll
LoadedModule[32]=C:\WINDOWS\System32\shcore.dll
LoadedModule[33]=C:\WINDOWS\System32\profapi.dll
LoadedModule[34]=C:\WINDOWS\System32\WS2_32.dll
LoadedModule[35]=C:\WINDOWS\SYSTEM32\AVICAP32.dll
LoadedModule[36]=C:\WINDOWS\SYSTEM32\WINMM.dll
LoadedModule[37]=C:\WINDOWS\SYSTEM32\MSVFW32.dll
LoadedModule[38]=C:\WINDOWS\SYSTEM32\WINMMBASE.dll
LoadedModule[39]=C:\WINDOWS\SYSTEM32\SSPICLI.DLL
LoadedModule[40]=C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_5.82.14393.447_none_0d5aa7fbb6d35646\COMCTL32.dll
LoadedModule[41]=C:\WINDOWS\System32\IMM32.DLL
LoadedModule[42]=C:\WINDOWS\SYSTEM32\d3d11.dll
LoadedModule[43]=C:\WINDOWS\SYSTEM32\dxgi.dll
LoadedModule[44]=C:\WINDOWS\System32\DriverStore\FileRepository\nvaowu.inf_amd64_a0bc429a357c29fa\nvwgf2umx.dll
LoadedModule[45]=C:\WINDOWS\SYSTEM32\VERSION.dll
LoadedModule[46]=C:\WINDOWS\SYSTEM32\bcrypt.dll
LoadedModule[47]=C:\WINDOWS\system32\uxtheme.dll
LoadedModule[48]=C:\WINDOWS\System32\clbcatq.dll
LoadedModule[49]=C:\WINDOWS\system32\dataexchange.dll
LoadedModule[50]=C:\WINDOWS\system32\dcomp.dll
LoadedModule[51]=C:\WINDOWS\system32\twinapi.appcore.dll
LoadedModule[52]=C:\WINDOWS\system32\explorerframe.dll
LoadedModule[53]=C:\WINDOWS\SYSTEM32\mscms.dll
LoadedModule[54]=C:\WINDOWS\SYSTEM32\USERENV.dll
LoadedModule[55]=C:\Windows\System32\msxml6.dll
LoadedModule[56]=C:\WINDOWS\System32\MMDevApi.dll
LoadedModule[57]=C:\WINDOWS\System32\PROPSYS.dll
LoadedModule[58]=C:\WINDOWS\System32\DEVOBJ.dll
LoadedModule[59]=C:\WINDOWS\SYSTEM32\AUDIOSES.DLL
LoadedModule[60]=C:\WINDOWS\SYSTEM32\wintypes.dll
LoadedModule[61]=C:\WINDOWS\System32\MSCTF.dll
LoadedModule[62]=C:\Users\-\Documents\d3dcompiler_43.dll
State[0].Key=Transport.DoneStage1
State[0].Value=1
FriendlyEventName=Stopped working
ConsentKey=APPCRASH
AppName=mpv
AppPath=C:\Users\-\Documents\mpv.exe
NsPartner=windows
NsGroup=windows8
ApplicationIdentity=45834744B05A35C8E22C56B2596F8141
MetadataHash=1348318049

**Comment by pennyscan** [comment]
finally solved...

If I add the option -- geometry=1600x1200 to the run command

this stops the frequent crashing  (1 in 10)

and also allows dsize to adjust the width without changing the height..

**Comment by garoto** [comment]
As merely a mpv user on Windows, I'd like to point two things (grains of salt needs are implied):

1.) You probably should replace `add` in your `vf` invocation with `set`, since you're not adding anything to the `vf` filter chain, just replacing the same existing filter. Possible explanation for the crashes you were experiencing perhaps? I tried to reproduce your exact input.conf contents and my mpv didn't crashed, so who knows.

2.) `dsize` is a deprecated filter and will probably be removed sometime in the near future, and after reading its description, I'm not even sure it applies to what you are trying to achive. It looks like a filter designed more for people running mpv on a certain display and outputting to a different display altogether, like a projector. I coudn't figure it out in any case. 

And last, take a look at ffmpeg video filters (https://ffmpeg.org/ffmpeg-filters.html) and try to find a better replacement for your needs there because all bult-in audio and video filters are being deprecated as far as I know.

**Comment by pennyscan** [comment]
Thanks for help.

Basically, I just need a key to reduce width by 1% each time I press it, I've spent many days on this problem without success


I can use ffmpeg like this.                      ( iw means input width in ffmpeg)

ffmpeg -i test.jpg -vf scale=iw*0.5:ih out.jpg       To reduce width and it works ok

But in mpv input.conf

1 vf set scale=iw*0.5:ih                    I get changing filters failed message

Is there a different way to use ffmpeg filters in input.conf

Regards
Keith hoyes







On 27 Jul 2017, at 11:16 pm, garoto <notifications@github.com<mailto:notifications@github.com>> wrote:


As merely a mpv user on Windows, I'd like to point two things (grains of salt needs are implied):

1.) You probably should replace add in your vf invocation with set, since you're not adding anything to the vf filter chain, just replacing the same existing filter. Possible explanation for the crashes you were experiencing perhaps? I tried to reproduce your exact input.conf contents and my mpv didn't crashed, so who knows.

2.) dsize is a deprecated filter and will probably be removed sometime in the near future, and after reading its description, I'm not even sure it applies to what you are trying to achive. It looks like a filter designed more for people running mpv on a certain display and outputting to a different display altogether, like a projector. I coudn't figure it out in any case.

And last, take a look at ffmpeg video filters (https://ffmpeg.org/ffmpeg-filters.html) and try to find a better replacement for your needs there because all bult-in audio and video filters are being deprecated as far as I know.
